uspace/lib/cpp/include/__bits/memory/shared_payload.hpp

```diff
@@ -90,43 +77,42 @@ namespace std::aux
     {
         public:
             shared_payload(T* ptr, D deleter = D{})
-                : data_{ptr}, deleter_{deleter},
-                  refcount_{1}, weak_refcount_{1}
+                : data_{ptr}, deleter_{deleter}
             { /* DUMMY BODY */ }
 
             template<class... Args>
             shared_payload(Args&&... args)
                 : data_{new T{forward<Args>(args)...}},
-                  deleter_{}, refcount_{1}, weak_refcount_{1}
+                  deleter_{}
             { /* DUMMY BODY */ }
 
             template<class Alloc, class... Args>
             shared_payload(allocator_arg_t, Alloc alloc, Args&&... args)
                 : data_{alloc.allocate(1)},
-                  deleter_{}, refcount_{1}, weak_refcount_{1}
+                  deleter_{}
             {
                 alloc.construct(data_, forward<Args>(args)...);
             }
 
             template<class Alloc, class... Args>
             shared_payload(D deleter, Alloc alloc, Args&&... args)
                 : data_{alloc.allocate(1)},
-                  deleter_{deleter}, refcount_{1}, weak_refcount_{1}
+                  deleter_{deleter}
             {
                 alloc.construct(data_, forward<Args>(args)...);
             }
 
             void destroy() override
             {
-                if (refs() == 0)
+                if (this->refs() == 0)
```

Comment:
Exactly, had to be added in several places as the template shared_payload now inherits from a non-templated type refcount_obj (because I wanted to move that logic out of smart pointers and reuse it in futures and promises).